### PR TITLE
Enable machine e2e tests for WSL

### PR DESCRIPTION
**Pull request by baude**
this pr allows us to run the machine e2e tests with the wsl provider.

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/machine/e2e/machine_test.go [hunk 1]
```diff
@@ -55,13 +57,23 @@ var _ = BeforeSuite(func() {
...
+			if err != nil {
+				Fail("unable to get virtual machine image")
+			}
+			downloadLocation = fcd.Location
+		} else {
+			fd, err := wsl.NewFedoraDownloader(provider.VMType(), "", defaultStream.String())
+			if err != nil {
+				Fail("unable to get WSL virtual image")
+			}
+			downloadLocation = fd.Get().URL.String()
```
Comment (Luap99) [inline pkg/machine/e2e/machine_test.go:73]: this needs to put put behind the windows build tag as the wsl backend can only be compiled on windows.

File: pkg/machine/e2e/README.md [hunk 2]
```diff
@@ -12,8 +12,15 @@
...
+1. $env:CONTAINERS_MACHINE_PROVIDER="hyperv"
+1. $env:MACHINE_IMAGE="https://fedorapeople.org/groups/podman/testing/hyperv/fedora-coreos-38.20230830.dev.0-hyperv.x86_64.vhdx.zip"
+1. `./test/tools/build/ginkgo.exe -vv  --tags "remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp remote" -timeout=90m --trace --no-color  pkg/machine/e2e/. `
+
+Note: Add `--focus-file "basic_test.go" ` to only run basic test
+
+### WSL
+1. Open a powershell as a regular user
+1. Build and copy win-sshproxy into bin/
+1. `./test/tools/build/ginkgo.exe -vv  --tags "remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp remote" -timeout=90m --trace --no-color  pkg/machine/e2e/. `
```
Comment (vrothberg) [inline pkg/machine/e2e/README.md:24]: A make target would be nice.
`make test-machine` or `make test-machine-wsl`?

File: pkg/machine/e2e/config_unix_test.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd
+
+package e2e_test
+
+import "github.com/containers/podman/v4/pkg/machine"
```
Comment (vrothberg) [inline pkg/machine/e2e/config_unix_test.go:5]: Also needs to import "github.com/onsi/ginkgo/v2" to see `Fail`. Same in the _windows file.

File: pkg/machine/e2e/README.md (hunk 2 above)
Comment (baude) [reply pkg/machine/e2e/README.md:24]: see #20028

**Comment by baude** [comment]
ok, ready for merge!

**Review by vrothberg — APPROVED** [review]
LGTM

**Pull request merged**